### PR TITLE
[Pathways] [Vitals] check app metadata routes in backend

### PR DESCRIPTION
**Pull request by danawillow**
## Description of the change

Added logic to read the appMetadata and validate that either:
* the user's state code is recidiviz
* the user has a routes object that contains a valid value for the resource they're trying to access

I did it for pathways and vitals- there are other routes like `community_projections`, `community_practices`, and `facilities_projections` where I couldn't figure out whether they had a backend API call that aligned with that route, and there were other API endpoints (`newRevocations`, `goals`, `community/explore`, `facilities/explore`, `projections/methodology.pdf`, `generateFileLink`) that I wasn't sure whether they had a matching route to check.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

https://github.com/Recidiviz/zenhub-tasks/issues/911

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by jovergaag** [comment]
> I did it for pathways and vitals- there are other routes like community_projections, community_practices, and facilities_projections where I couldn't figure out whether they had a backend API call that aligned with that route, and there were other API endpoints (newRevocations, goals, community/explore, facilities/explore, projections/methodology.pdf, generateFileLink) that I wasn't sure whether they had a matching route to check.

- `community_projections` and `facilities_projections` are no longer valid - once US_ID moved over to Pathways the projections are folded inside of the Pathways permissions.
- `community_practices` hits the `vitals` endpoint for ND, from the Core dashboard.
- All of the others you listed do not have permissions controlling them, all users have access to the whole page and all features on the page.

File: server/routes/api.js [hunk 1]
```diff
@@ -183,6 +195,14 @@ function populationProjectionsMethodology(req, res) {
 }
 
 function vitals(req, res) {
+  const appMetadata = getAppMetadata(req);
+  const allowed =
+    appMetadata.state_code === "recidiviz" || appMetadata.routes?.operations;
```
Comment (jovergaag) [inline server/routes/api.js:200]: This should also include `community_practices`

**Review by jovergaag — CHANGES_REQUESTED** [review]
Looks good, just need to add the ND vitals permission!

File: server/routes/api.js (hunk 1 above)
Comment (danawillow) [reply server/routes/api.js]: Thanks for the info! Done.

**Review by jovergaag — APPROVED** [review]

**Pull request merged**